imitation/tests/test_tabular.py

```diff
@@ -0,0 +1,261 @@
+            [0, 0, 0, 0.05, 0.95],
+            # action 1: goes to state 3 (bad)
+            [0, 0, 0, 1, 0],
+            # action 2: goes to state 4 (good!)
+            [0, 0, 0, 0, 1]
+        ],
+        # transitions out of state 2 (basically the same)
+        [
+            # action 0: goes to state 3 or 4 (sub-optimal)
+            [0, 0, 0, 0.05, 0.95],
+            # action 1: goes to state 3 (bad)
+            [0, 0, 0, 1, 0],
+            # action 2: goes to state 4 (good!)
+            [0, 0, 0, 0, 1]
+        ],
+        # transitions out of state 3 (all go to state 4)
+        [
+            # action 0
+            [0, 0, 0, 0, 1],
+            # action 1
+            [0, 0, 0, 0, 1],
+            # action 2
+            [0, 0, 0, 0, 1]
+        ],
+        # transitions out of state 4 (all go back to state 0)
+        [
+            # action 0
+            [1, 0, 0, 0, 0],
+            # action 1
+            [1, 0, 0, 0, 0],
+            # action 2
+            [1, 0, 0, 0, 0]
+        ],
+    ])
+    reward_matrix = np.array([
+        # state 0 (okay reward, but we can't go back so it doesn't matter)
+        1,
+        # states 1 & 2 have same (okay) reward
+        2,
+        2,
+        # state 3 has very negative reward (so avoid it!)
+        -20,
+        # state 4 has pretty good reward (good enough that we should move out
+        # of 1 & 2)
+        3
+    ])
+    horizon = 20
+
+
+def test_policy_om_reasonable_mdp():
+    # MDP described above
+    mdp = ReasonableMDP()
+    # get policy etc. for our MDP
+    V, Q, pi = mce_partition_fh(mdp)
+    Dt, D = mce_occupancy_measures(mdp, pi=pi)
+    assert np.all(np.isfinite(V))
+    assert np.all(np.isfinite(Q))
+    assert np.all(np.isfinite(pi))
+    assert np.all(np.isfinite(Dt))
+    assert np.all(np.isfinite(D))
+    # check that actions 0 & 2 (which go to states 1 & 2) are roughly equal
+    assert np.allclose(pi[:19, 0, 0], pi[:19, 0, 2])
+    # also check that they're by far preferred to action 1 (that goes to state
+    # 3, which has poor reward)
+    assert np.all(pi[:19, 0, 0] > 2 * pi[:19, 0, 1])
+    # make sure that states 3 & 4 have roughly uniform policies
+    pi_34 = pi[:5, 3:5]
+    assert np.allclose(pi_34, np.ones_like(pi_34) / 3.0)
+    # check that states 1 & 2 have similar policies to each other
+    assert np.allclose(pi[:19, 1, :], pi[:19, 2, :])
+    # check that in state 1, action 2 (which goes to state 4 with certainty) is
+    # better than action 0 (which only gets there with some probability), and
+    # that both are better than action 1 (which always goes to the bad state).
+    assert np.all(pi[:19, 1, 2] > pi[:19, 1, 0])
+    assert np.all(pi[:19, 1, 0] > pi[:19, 1, 1])
+
+    # now compute demonstrator features
+    # initial reward weights
+    rmodel = LinearRewardModel(mdp.obs_dim, seed=13)
+    opt = AMSGrad(rmodel, alpha_sched=1e-2)
+    final_weights, final_counts = maxent_irl(
```

Comment:
I like the test case up until here (cleverly constructed MDP). Can we do anything to sanity check `maxent_irl`?
Also inclined to split these last few lines into a separate test case since they don't seem to depend on the previous part. And make it a parameterized test, so we test different reward models and optimizers. We should ideally test all combinations for a simple "smoke test" (i.e. does it run), maybe only test combinations we expect to get good results more extensively.